packages/core/echo/echo/src/experimental/query-2.test.ts

```diff
@@ -0,0 +1,58 @@
+import { Schema } from 'effect';
+import { Type } from '..';
+
+declare const Relation: any;
+declare const Query: any;
+
+//
+// Example schema
+//
+
+const Person = Schema.Struct({
+  name: Schema.String,
+}).pipe(Type.def({ typename: 'dxos.org/type/Person', version: '0.1.0' }));
+
+const Org = Schema.Struct({
+  name: Schema.String,
+}).pipe(Type.def({ typename: 'dxos.org/type/Org', version: '0.1.0' }));
+
+const WorksFor = Schema.Struct({
+  since: Schema.String,
+}).pipe(Relation.def({ typename: 'dxos.org/type/WorksFor', version: '0.1.0', source: Person, target: Org }));
+
+const Task = Schema.Struct({
+  title: Schema.String,
+  createdAt: Schema.String,
+  assignee: Type.Ref(Person),
+}).pipe(Type.def({ typename: 'dxos.org/type/Task', version: '0.1.0' }));
+
+//
+// Example queries
+//
+
+// Query<Person>
+const getAllPeople = Query.type(Person);
+
+// Query<Person>
+const getAllPeopleNamedFred = Query.type(Person, { name: 'Fred' });
+
+// Query<Org>
+declare const fred: any;
+const getAllOrgsFredWorkedForSince2020 = Query.type(Person, { id: fred.id })
+  .outgoingRelations(WorksFor, { since: Query.gt('2020') })
+  .targets();
+
+// Query<Task>
+const getAllTasksForFred = Query.type(Person, { id: fred.id }).referencedBy(Task, 'assignee');
+
+// Query<Task>
+const allTasksForEmployeesOfCyberdyne = Query.type(Org, { name: 'Cyberdyne' })
+  .incomingRelations(WorksFor)
+  .sources()
```

Comment:
would you ever want the `targets` of `incomingRelations`? probably not because that's what you already have? this feels like it could just be folded into the `incomingRelations` api?

Reply:
The reason those exist is that there are cases where you want to query just for relation target, i.e. traverse the relation in the query. 

But there are cases where you want the relation itself to be returned, for example.

```ts
// build my resume

const { results } = await db.query(Query.type(Person, { id: me.id })).sourceOf(WorksFor)).run(); // Query<WorksFor>

results.map(result => ({
   orgName: getTarget(result).name,
   since: result.since,
   until: result.until,
}))
```

Reply:
I'd say those should be different methods:
- `sourceOf` traverses the relation via the source
- `targetOf` traverses the relation via the target
- `relationOf` queries for the relation itself (perhaps with options to specify if you want to limit to only the source or target)